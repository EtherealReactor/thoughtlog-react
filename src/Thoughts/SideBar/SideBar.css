/* .SidebarNav {
  min-width: 25%;
} */

.SidebarNav {
  /* display: flex; */
}

.SidebarList {
  list-style: none;
  display: flex;
  padding: 0;
  text-align: center;
  margin: 0;
  cursor: pointer;
}

.SideBarListItem {
  padding: .5rem;
  background: #F0F0F0;
  margin: 0 0.5rem;;
  border-radius: 10px;
}

.SideBarLink:link,
.SideBarLink:visited,
.SideBarLink:hover,
.SideBarLink:active {
  color: #A0A3A5;
  font-size: 1.2rem;
  text-decoration: none;
  font-weight: bold;
}

.SideBarIcon {
  margin-left: 1rem;
  color: #895EA8;
}