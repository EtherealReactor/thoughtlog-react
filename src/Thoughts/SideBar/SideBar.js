import React from 'react';
import { Link } from 'react-router-dom';
import * as FontAwesome from 'react-icons/lib/fa';

import SidebarStyles from './SideBar.css';

const SideBar = (props) => {
  return (
    <aside className={SidebarStyles.SidebarNav}>
      <nav>
        <ul className={SidebarStyles.SidebarList}>
          <li className={SidebarStyles.SideBarListItem}>
            <Link to="/thoughts/new" className={SidebarStyles.SideBarLink}>Quick Thought
              <FontAwesome.FaPlusSquareO className={SidebarStyles.SideBarIcon}/>
            </Link>
          </li>
          <li className={SidebarStyles.SideBarListItem}>
            <Link to="/thoughts/drafts" className={SidebarStyles.SideBarLink} >Saved Thought
              <FontAwesome.FaArrowDown className={SidebarStyles.SideBarIcon}/>
            </Link>
          </li>
        </ul>
      </nav>    
    </aside>
  );
}

export default SideBar;