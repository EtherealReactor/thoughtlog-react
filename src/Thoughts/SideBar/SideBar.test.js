import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import SideBar from './SideBar';

describe("Component === Thoughts === Sidebar", () => {
  it("must match snapshot", () => {
    let wrapper = shallow(<SideBar />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should contain all the necessary dom", () => {
    let wrapper = shallow(<SideBar />);
    expect(wrapper.find('aside')).toHaveLength(1);
    expect(wrapper.find('nav')).toHaveLength(1);
    expect(wrapper.find('ul')).toHaveLength(1);
    expect(wrapper.find('li')).toHaveLength(2);
    expect(wrapper.find('Link')).toHaveLength(2);
  });
});