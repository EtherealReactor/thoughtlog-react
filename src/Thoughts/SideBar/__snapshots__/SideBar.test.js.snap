// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component === Thoughts === Sidebar must match snapshot 1`] = `
<aside>
  <nav>
    <ul>
      <li>
        <Link
          replace={false}
          to="#"
        >
          Quick Thought
          <FaPlusSquareO />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/thoughts/drafts"
        >
          Saved Thought
          <FaArrowDown />
        </Link>
      </li>
    </ul>
  </nav>
</aside>
`;
